packages/components/namespace-store/namespace-store-form.tsx

```diff
@@ -81,42 +81,61 @@ const NamespaceStoreForm: React.FC<NamespaceStoreFormProps> = withHandlePromise<
   NamespaceStoreFormProps & HandlePromiseProps
 >((props) => {
   const { t } = useTranslation();
-  const [nsName, setNsName] = React.useState('');
   const [provider, setProvider] = React.useState(BC_PROVIDERS.AWS);
-  const [pvc, setPVC] = React.useState(null);
-  const [folderName, setFolderName] = React.useState('');
-  const [providerDataState, providerDataDispatch] = React.useReducer(
-    providerDataReducer,
-    initialState
+  const [formDataState, formDataDispatch] = React.useReducer(
+    nssFormReducer,
+    initialStateFormCreation
   );
-  const handleNsNameTextInputChange = (strVal: string) => setNsName(strVal);
+  const handleNsNameTextInputChange = (strVal: string) =>
+    formDataDispatch({ type: 'setName', value: strVal });
   const {
     onCancel,
     className,
     inProgress,
-    errorMessage,
     handlePromise,
     redirectHandler,
     namespace,
+    errorMessage,
   } = props;
 
+  const isFormValid = () => {
+    const {
+      name,
+      accessKey,
+      secretKey,
+      secretName,
+      region,
+      endpoint,
+      target,
+      pvc,
+      pvcFolderName,
+    } = formDataState;
+    return (
+      name &&
+      name.length <= 42 &&
+      (((region || endpoint) &&
+        (secretName || (secretKey && accessKey)) &&
+        target) ||
+        (pvc && pvcFolderName))
+    );
+  };
```

Comment:
I assume making things required and create button being part of same formgroup should disable the create. This doesn't look like a good solution to me.

@bipuladh @GowthamShanmugam Thoughts?